### PR TITLE
Improve federated init, validate reactor program, delimited proto messages

### PR DESCRIPTION
**Pull request by erlingrj**
This PR got a little fat...

- Improve setup of connection. Now it is done in Environment at assemble. So no need to manually code that.
- Fixed some small mistakes in TcpIpChannel receiving if multiple messages were there in the buffer
- Use delimted serialize/deserialize
- Fix a typo in posix examples
- Add a function that validates that the program is correctly set up

### CODE REVIEW

File: src/environment.c [hunk 1]
```diff
@@ -6,7 +6,55 @@
...
+          all_connected = false;
+          break;
+        default:
+          throw("Could not connect to federate during assemble");
+          break;
+        }
+      }
+    }
+    if (!all_connected) {
+      self->platform->wait_for(self->platform, MSEC(10)); // TODO: Should  the 10msec be configurable?
```
[the remark this answers is not in the document]
Comment (erlingrj) [reply src/environment.c]: Thanks for the comment. I did not address it now as it requires some additions to the NetworkChannel API I think. I think we must think a little about it when we introduce more channels.

[the remark this answers is not in the document]
Comment (erlingrj) [reply src/environment.c]: It seems that you address this with your PR! Great.

**Comment by erlingrj** [comment]
@LasseRosenow, I am here providing a initial implementation of your additions to the channel API. I think we should merge this soon before it gets too fat

File: src/environment.c [hunk 2]
```diff
@@ -41,6 +42,9 @@ void Environment_assemble(Environment *self) {
           break;
         case LF_IN_PROGRESS:
         case LF_TRY_AGAIN:
+          if (chan->expected_try_connect_duration < wait_before_retry && chan->expected_try_connect_duration > 0) {
```
Comment (LasseRosenow) [inline src/environment.c:45]: Nice! This is what I was thinking also :)

**Comment by LasseRosenow** [comment]
Thank you for implementing the new `NetworkChannelState` enum for the `TcpIpChannel`. It looks good from what I can see.

**Pull request merged**